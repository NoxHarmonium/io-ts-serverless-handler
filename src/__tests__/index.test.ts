import { APIGatewayProxyEvent } from "aws-lambda";
import * as t from "io-ts";
import { codecHandler, HandlerFunction } from "..";

const mockSchema = {
  queryStringParameters: {
    pageSize: t.number
  },
  body: t.type({
    message: t.string
  })
};

const mockHandler: HandlerFunction<typeof mockSchema> = async ({
  queryStringParameters: { pageSize },
  body
}) => ({
  pageSize: pageSize + 2,
  body
});

describe("when wrapping a standard handler", () => {
  describe("when the payload is valid", () => {
    it("should succeed", () => {
      const mockEvent = ({
        queryStringParameters: {
          pageSize: 4
        },
        body: JSON.stringify({
          message: "hello"
        })
      } as unknown) as APIGatewayProxyEvent;
      const handler = codecHandler(mockSchema, mockHandler);

      const result = handler(mockEvent);
      expect(result).toMatchSnapshot();
    });
  });
  describe("when the payload is invalid", () => {
    it("should throw an error", () => {
      const mockEvent = ({
        queryStringParameters: {
          pageSize: 4
        },
        body: 5
      } as unknown) as APIGatewayProxyEvent;
      const handler = codecHandler(mockSchema, mockHandler);

      expect(() => handler(mockEvent)).toThrowError();
    });
  });
});
