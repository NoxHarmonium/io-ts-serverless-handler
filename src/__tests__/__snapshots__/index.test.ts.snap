// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when wrapping a standard handler when the payload is valid should succeed 1`] = `Promise {}`;
